tests/e2e/envfuncsext/nebulacluster.go

```diff
@@ -165,6 +166,19 @@ func WaitNebulaClusterReady(opts ...NebulaClusterOption) env.Func {
 				"generation", nc.Generation,
 			)
 
+			if nc.Status.Graphd.Phase == appsv1alpha1.RunningPhase && nc.Status.Metad.Phase == appsv1alpha1.RunningPhase && nc.Status.Storaged.Phase == appsv1alpha1.RunningPhase && first {
+				klog.V(4).Infof("First running phase detected for nc cluster. Trying again to see if nc cluster is really running")
+				first = false
+
+				if err = cfg.Client().Resources().Get(ctx, nc.GetName(), nc.GetNamespace(), nc); err != nil {
+					klog.ErrorS(err, "Get NebulaCluster failed", "namespace", nc.Namespace, "name", nc.Name)
+					return false, err
+				}
+				return false, nil
+			} else {
+				first = true
+			}
+
```

Comment:
Why add this piece of logic? What if not?

Reply:
Unable to reproduce in the latest operator version. Considering this fixed. This code has been removed.